region/maridia/inner-yellow/Kassiuz Room.json

```diff
@@ -111,15 +111,85 @@
       "requires": [
         {"resetRoom": {"nodes": [1]}},
         {"or": [
-          "Spazer",
-          "Wave",
-          "Plasma",
-          "canUseGrapple",
-          "ScrewAttack"
+          {"and": [
+            "Grapple",
+            {"cycleFrames": 180}
+          ]},
+          {"and": [
+            {"or": [
+              "Plasma",
+              "Wave",
+              "Spazer",
+              "ScrewAttack"
+            ]},
+            {"cycleFrames": 210}
+          ]},
+          {"and": [
+            "canFarmWhileShooting",
+            {"cycleFrames": 330}
+          ]},
+          {"cycleFrames": 400}
+        ]}
+      ],
+      "farmCycleDrops": [{"enemy": "Puyo", "count": 2}]
+    },
+    {
+      "link": [1, 1],
+      "name": "Puyo and Choot Farm",
+      "requires": [
+        {"or": [
+          {"resetRoom": {"nodes": [1]}},
+          {"and": [
+            {"resetRoom": {"nodes": [2]}},
+            {"cycleFrames": 120}
+          ]}
         ]},
-        {"partialRefill": {"type": "Energy", "limit": 200}},
-        {"partialRefill": {"type": "Missile", "limit": 12}},
-        {"partialRefill": {"type": "PowerBomb", "limit": 6}}
+        {"or": [
+          "canWalljump",
+          {"and": [
+            "SpaceJump",
+            {"cycleFrames": 300}
+          ]},
+          {"and": [
+            "canTrickySpringBallJump",
+            {"cycleFrames": 520},
+            "h_doubleEquipmentScreenCycleFrames",
+            "h_doubleEquipmentScreenCycleFrames",
+            "h_doubleEquipmentScreenCycleFrames"
+          ]}
+        ]},
+        "canDodgeWhileShooting",
+        {"or": [
+          {"and": [
+            "Plasma",
+            {"cycleFrames": 820}
+          ]},
+          {"and": [
+            "Grapple",
+            {"cycleFrames": 900}
+          ]},
+          {"and": [
+            "ScrewAttack",
+            {"cycleFrames": 1000}
+          ]},
+          {"and": [
+            "Wave",
+            {"cycleFrames": 1080}
+          ]},
+          {"and": [
+            "Spazer",
+            {"cycleFrames": 1140}
+          ]},
+          {"cycleFrames": 2040}
+        ]},
+        {"and": [
```

Comment:
I thought tests caught top-level `and` now

Reply:
Good catch. Looks like the tests checked `and` inside of `and` but not `and` at the top level. Added that now. There weren't too many violations so I went through and fixed them all.